parser/impl/parser.cpp

```diff
@@ -633,74 +678,57 @@ sPtr<ast::Expression> Parser::parseBinaryExpression(int prec1) {
         op = m_lexer.Next();
 
         ast::BinaryExpression expr;
+        if (lhs != nullptr) {
```

Comment:
I didn't want to break the code already written (and a change like that might), while also adding the information if possible without causing a segmentation fault in case of some errors.